### PR TITLE
test: add on_payload_attestation_message fork choice test handler

### LINKED ISSUE #9364: Add handler for new `on_payload_attestation_message` tests
Need to add new handler to our fork choice tests for `on_payload_attestation_message`, see https://github.com/ethereum/consensus-specs/pull/5206

See [Nightly Spec Tests run](https://github.com/ChainSafe/lodestar/actions/runs/25783769058)

### PR DESCRIPTION
**Pull request by markolazic01**
**Motivation**

The consensus-specs repo added new fork choice test vectors for on_payload_attestation_message (ethereum/consensus-specs#5206). Lodestar's spec test runner needs a handler for this step type so the nightly spec tests can run these vectors instead of failing on an unrecognized step.

**Description**

Adds test infrastructure to handle `on_payload_attestation_message` steps in fork choice spec tests:

 - OnPayloadAttestationMessage step type and `isPayloadAttestationMessage` type guard
 - SSZ fixture loader for `payload_attestation_message_*.ssz_snappy` files
 - Step handler that deserializes the message and calls `notifyPtcMessages`

Closes #9364

**AI Assistance Disclosure**

- [x] External Contributors: I have read the [contributor guidelines](https://github.com/ChainSafe/lodestar/blob/unstable/CONTRIBUTING.md#ai-assistance-notice) and disclosed my usage of AI below.

Claude was used for codebase navigation, understanding the spec test infrastructure, reviewing diffs for bugs, and drafting the PR description. All code was authored manually.

### CODE REVIEW

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 1]
```diff
@@ -201,6 +203,27 @@ const forkChoiceTest =
...
+                root: step.payload_attestation_message,
+                valid: Boolean(step.valid),
+              });
+              const payloadAttestationMessage = testcase.payloadAttestationMessages.get(
+                step.payload_attestation_message
+              );
+              if (!payloadAttestationMessage)
+                throw Error(`No payload attestation message ${step.payload_attestation_message}`);
+              const blockRoot = toRootHex(payloadAttestationMessage.data.beaconBlockRoot);
+              // TODO: notifyPtcMessages expects PTC-relative indices (0-15), but the spec test
```
Comment (markolazic01) [inline packages/beacon-node/test/spec/presets/fork_choice.test.ts]: Hey @nflaig, apologies for bothering, just wanted to ask if there is a helper function i can use for this index conversion, or if you would like me to add it somewhere? I got a little stuck on that part and thought it is best to ask you first.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 2]
```diff
@@ -201,6 +203,27 @@ const forkChoiceTest =
...
+              // provides global validator indices. Need to resolve via PTC committee lookup.
+              chain.forkChoice.notifyPtcMessages(
+                blockRoot,
+                [payloadAttestationMessage.validatorIndex],
+                payloadAttestationMessage.data.payloadPresent
+              );
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: Comment is in context with the `TODO` but it seems that `getIndicesInPayloadTimelinessCommittee` doesn't exist in the codebase.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 1 above)
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: Hey @wemeetagain, apologies for bothering, any chance you can give me a hint on this?

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 3]
```diff
@@ -201,6 +203,38 @@ const forkChoiceTest =
...
+                valid: isValid,
+              });
+              const payloadAttestationMessage = testcase.payloadAttestationMessages.get(
+                step.payload_attestation_message
+              );
+              if (!payloadAttestationMessage)
+                throw Error(`No payload attestation message ${step.payload_attestation_message}`);
+              try {
+                const blockRoot = toRootHex(payloadAttestationMessage.data.beaconBlockRoot);
+                const ptc = cachedState.epochCtx.getPayloadTimelinessCommittee(chain.getHeadState().slot);
```
Comment (markolazic01) [inline packages/beacon-node/test/spec/presets/fork_choice.test.ts]: I'm using `chain.getHeadState().slot` for the PTC committee lookup rather than `payloadAttestationMessage.data.slot`. Please let me know if that is alright.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 4]
```diff
@@ -201,6 +203,38 @@ const forkChoiceTest =
...
+                const ptcIndex = ptc.indexOf(payloadAttestationMessage.validatorIndex);
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: While not practically reproducible on mainnet, I do think this is a possibility in devnets. I don't know if there are any test vectors that stress test this.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 3 above)
Comment (wemeetagain) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: Why?

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 5]
```diff
@@ -201,6 +203,38 @@ const forkChoiceTest =
...
+                if (ptcIndex === -1) {
+                  throw Error(
+                    `Validator ${payloadAttestationMessage.validatorIndex} not in PTC for slot ${payloadAttestationMessage.data.slot}`
+                  );
+                }
+                chain.forkChoice.notifyPtcMessages(
+                  blockRoot,
+                  [ptcIndex],
+                  payloadAttestationMessage.data.payloadPresent
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: - I think this is implemented now after https://github.com/ChainSafe/lodestar/pull/9387

[the remark this answers is not in the document]
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: thanks, adapted it

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 3 above)
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: It was improper, now on `payloadAttestationMessage.data.slot`.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 6]
```diff
@@ -201,6 +203,38 @@ const forkChoiceTest =
...
+                );
+                if (!isValid) throw Error("Expected invalid payload attestation message to throw");
+              } catch (e) {
+                if (isValid) throw e;
```
[the remark this answers is not in the document]
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: should be good now

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts]: this is resolved as well with the usage of `getIndicesInPayloadTimelinessCommittee`

**Review by wemeetagain — COMMENTED** [review]
lgtm, lets see it run in CI

**Comment by markolazic01** [comment]
Based on the failing log I assume that the `notifyPtcMessages` function should be adapted or a wrapper should be introduced with full spec validation.

Following negative tests fail:
`not_ptc_member`: `ptcIndices` is empty in this particular case, and `notifyPtcMessages` does nothing instead of throwing
`current_slot_and_signature`: no BLS signature verification, `notifyPtcMessages` doesn't receive or check the signature at all

I am happy to manage this here or do it in a separate PR, please let me know what you prefer.

Log: 

>  FAIL   spec-minimal  packages/beacon-node/test/spec/presets/fork_choice.test.ts > gloas/fork_choice/on_payload_attestation_message/pyspec_tests > gloas/fork_choice/on_payload_attestation_message/pyspec_tests/on_payload_attestation_message_current_slot_and_signature
 FAIL   spec-minimal  packages/beacon-node/test/spec/presets/fork_choice.test.ts > gloas/fork_choice/on_payload_attestation_message/pyspec_tests > gloas/fork_choice/on_payload_attestation_message/pyspec_tests/on_payload_attestation_message_not_ptc_member
Error: Expect error since this is a negative test

**Comment by nflaig** [comment]
> Based on the failing log I assume that the `notifyPtcMessages` function should be adapted or a wrapper should be introduced with full spec validation.

a wrapper might be required, but also we shouldn't be passing these without https://github.com/ChainSafe/lodestar/pull/9425 unless there is no spec tests for this, in that case, we should add a new spec test for it

**Comment by markolazic01** [comment]
> also we shouldn't be passing these without https://github.com/ChainSafe/lodestar/pull/9425 unless there is no spec tests for this, in that case, we should add a new spec test for it

The spec test for this exists [here](https://github.com/ethereum/consensus-specs/blob/6370819a35e9558822ef024126cc09ee3666827d/tests/core/pyspec/eth_consensus_specs/test/gloas/fork_choice/test_on_payload_attestation_message.py#L98). 

It was already passing by accident as the PTC lookup returned empty indices for the mismatched slot, so `notifyPtcMessages` silently did nothing, producing the correct outcome for the wrong reason. 

With #9425 merged it now passes correctly via the slot check.

Will continue on with building a wrapper.

**Comment by markolazic01** [comment]
merged to resolve a conflict

**Comment by markolazic01** [comment]
Incorporated improvements from #9445, made by @twoeths:

- Vote check assertions for payload_timeliness_vote and payload_data_availability_vote in the `Checks` step
- BLS try/catch for malformed signature deserialization
- `isStatePostGloas` guard
- `clock.currentSlot` instead of manual computation
- Slot mismatch as an `if` wrapper instead of `continue` (nit)
- Added `getPayloadTimelinessVotes` / `getPayloadDataAvailabilityVotes` on `forkChoice` and `protoArray`

**Review by twoeths — DISMISSED** [review]
lgtm

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 7]
```diff
@@ -201,6 +206,87 @@ const forkChoiceTest =
...
+
+                  if (ptcIndices.length === 0) {
+                    throw Error(`Validator ${payloadAttestationMessage.validatorIndex} is not a member of the PTC`);
+                  }
+
+                  if (clock.currentSlot !== payloadAttestationMessage.data.slot) {
+                    throw Error(
+                      `Message slot ${payloadAttestationMessage.data.slot} is not current slot ${clock.currentSlot}`
+                    );
+                  }
```
Comment (nflaig) [inline packages/beacon-node/test/spec/presets/fork_choice.test.ts:245]: why do we need this?

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 8]
```diff
@@ -201,6 +206,87 @@ const forkChoiceTest =
...
+                  if (!signatureValidity) throw Error("Invalid payload attestation signature");
+
+                  chain.forkChoice.notifyPtcMessages(
+                    blockRoot,
+                    payloadAttestationMessage.data.slot,
+                    ptcIndices,
+                    payloadAttestationMessage.data.payloadPresent,
+                    payloadAttestationMessage.data.blobDataAvailable
+                  );
```
Comment (nflaig) [inline packages/beacon-node/test/spec/presets/fork_choice.test.ts:275]: curious, did you remove checks in our implementation of `notifyPtcMessages` to see if the spec tests fail?

we should make sure there is (1) no gap in how we set up the test harness that hides test coverage of the actual code (that isn't strictly needed, like the signature verification) and (2) that spec tests are complete and cover all edge cases

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 7 above)
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts:245]: The `current_slot_and_signature` test has a PAM step at slot 2 with `data.slot=1`, which expects rejection. So this check helps us throw on a stale message.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 8 above)
Comment (markolazic01) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts:275]: I did that through the development, but it is probably a good idea to do it again systematically. Will write here if I find anything suspicious.

**Comment by lodekeeper** [comment]
**Coverage pressure-test results** — followup to #discussion_r3346890899 (`notifyPtcMessages` + handler).

Methodology: removed each guard one at a time (always rebuilt `@lodestar/fork-choice` between proto-array changes), reran `gloas/fork_choice/on_payload_attestation_message/*` minimal vectors (7 total: `current_slot_and_signature`, `from_block`, `multiple_ptc_members_vote_independently`, `not_ptc_member`, `slot_mismatch`, `unknown_block_root`, `valid`), recorded outcome, restored. All commit-locked to `89edd00`.

### Coverage map

#### Real-code — `protoArray.notifyPtcMessages` (`packages/fork-choice/src/protoArray/protoArray.ts:679-709`)

| # | Check | Lines | Removed → outcome | Verdict |
|---|---|---|---|---|
| A1 | `votes/attended/daVotes === undefined → return` | 689-692 | 7/7 pass | **Unverified.** Harness H1 throws "Block not found" for unknown roots before impl is reached. Non-Gloas root case is untestable through Gloas anchor state. |
| A2 | `node.slot !== slot → return` | 694-699 | 7/7 pass | **Unverified.** Harness H2 gates impl call on `protoBlock.slot === data.slot`. A2 matches `notify_ptc_messages` spec (`store.get_block_by_root(block_root).slot != slot: return`) and is the only safeguard on the from-block path (`importBlock.ts:264` calls `notifyPtcMessages` without pre-validating PA slot vs block slot). **Potential spec-test gap** — see below. |
| A3 | `ptcIndex < 0 || >= PTC_SIZE → throw` | 701-704 | 7/7 pass | **Unreachable in practice.** Gossip/API resolve indices via `getIndicesInPayloadTimelinessCommittee` (returns `[0, PTC_SIZE)`); from-block derives from `PayloadAttestation.aggregationBits` (SSZ `Bitvector[PTC_SIZE]`). Defense-in-depth, not testable from spec vectors. |

#### Test-harness handler (`packages/beacon-node/test/spec/presets/fork_choice.test.ts:218-287`)

| # | Check | Lines | Removed → outcome | Verdict |
|---|---|---|---|---|
| H1 | `!protoBlock → throw "Block not found"` | 224-226 | `unknown_block_root` fails (H8 sentinel re-throw) | **Harness scaffolding.** Impl A1 would silently return; post-step `checks` verifies vote arrays unchanged. Without H1+H8 the test would pass via the post-check. |
| H2 | `if (protoBlock.slot === data.slot) { ... }` gate | 228 | `slot_mismatch` fails at H4 (`Validator … is not a member of the PTC`) | **Load-bearing.** Removing it doesn't expose impl A2 — it cascades into downstream harness checks (H4 fires because PTC lookup for the mismatched slot returns empty). Effectively mirrors impl A2's silent-return semantics. |
| H3 | `!isStatePostGloas → throw` | 236 | 7/7 pass | **Never fires.** All current vectors use post-Gloas anchor state. Sanity assert only. |
| H4 | `ptcIndices.length === 0 → throw` | 243-245 | `not_ptc_member` fails (H8 sentinel re-throw) | **Harness scaffolding.** Impl's loop is a no-op on empty `ptcIndices`; post-check verifies state unchanged. Without H4+H8 the test would pass. |
| H5 | `clock.currentSlot !== data.slot → throw` | 247-251 | `current_slot_and_signature` fails via post-check (votes set when they should be null) | **Does real validation.** Impl `notifyPtcMessages` doesn't check this. In real Lodestar, `validateGossipPayloadAttestationMessage` (`packages/beacon-node/src/chain/validation/payloadAttestationMessage.ts:42-48`) handles it via `isCurrentSlotGivenGossipDisparity`. Harness inlines this because the spec definition of `on_payload_attestation_message` includes it. |
| H6 | `!validatorPubkey → throw` | 254-256 | 7/7 pass | **Never fires.** All PAM validator indices in vectors are registered. |
| H7 | `!signatureValidity → throw` | 274 | `current_slot_and_signature` fails via post-check | **Does real validation.** Same pattern as H5 — impl `notifyPtcMessages` doesn't verify signature; real flow does it in `validateGossipPayloadAttestationMessage` (lines 125-135). Harness inlines because spec's `on_payload_attestation_message` mandates signature verification when `is_from_block == False`. |
| H8 | `if (!isValid) throw "Expect error…"` sentinel | 284 | 7/7 pass | **Redundant safety net.** The post-step `expect(getPayloadTimelinessVotes(...)).toEqualWithMessage(votes, …)` assertion at `checks` steps (lines 625-632) is the real correctness signal for negative tests. H8 is belt-and-suspenders on top of state verification. |

### Interpretation

The harness handler is effectively an inline re-implementation of the spec's `on_payload_attestation_message(store, payload_attestation_message, is_from_block)` — the parent function that wraps `notify_ptc_messages`. Specifically the checks map 1:1 to `validateGossipPayloadAttestationMessage` (current-slot, unknown-block, block-slot-match, PTC member, pubkey, signature). 

That answers Nico's concern (1) ("test harness setup hides test coverage of the actual code"):
- **H5 + H7 hide real-flow coverage of `validateGossipPayloadAttestationMessage`** — they replicate that function's logic inline. The spec test does not call `validateGossipPayloadAttestationMessage`, so any divergence in `validateGossipPayloadAttestationMessage`'s implementation (e.g. forgetting current-slot check, gossip-disparity tolerance, MAX_GOSSIP_CLOCK_DISPARITY off-by-one) wouldn't be caught. This is expected — these tests target the spec's `on_payload_attestation_message`, not lodestar's pre-fork-choice validation path — but it is a coverage gap from the impl's perspective.
- **H1 + H4 + H8 are redundant given post-checks** — they convert "impl silently no-ops" into a thrown error, but the `checks` step's vote-array `toEqualWithMessage` would catch any state divergence. Could be removed without losing correctness signal.
- **H3 + H6 are sanity asserts that never fire** under current vectors. Worth keeping as defensive but they're not exercised.

### Spec-test gaps (Nico's concern (2))

- **A2 is structurally unverified.** No spec vector exercises `notify_ptc_messages`'s `block.slot != slot` silent return at the `notify_ptc_messages` boundary. The closest is `slot_mismatch`, but it's intercepted by the H2 gate before reaching impl. Worth flagging upstream: a vector that drives `notify_ptc_messages` with mismatched slot via a `block` step containing a malformed `payload_attestations` body (a use case A2 directly protects against on `importBlock.ts:264`) would close this.
- **A3 is structurally untestable.** SSZ `Bitvector[PTC_SIZE]` + spec helper bound the input by construction. Not a spec gap, just unreachable defense.

I think A2 is worth raising in consensus-specs — willing to draft a vector if you want it, but it's likely a one-line spec-test addition (PA in block body with `data.slot != block.slot`) and may already be implicitly covered by the `from_block` flow if it tests that codepath. Let me know if I should draft it.

### Out of scope (per #on-payload-attestation-message-tests Discord 08:23 UTC): `validateGossipPtcMessage` coverage flag

The spec tests for `on_payload_attestation_message` essentially **shadow** `validateGossipPayloadAttestationMessage` — every check in the latter is mirrored by an inline harness check (H1↔unknown-block, H5↔current-slot, H2 condition↔block-slot-match, H3↔post-Gloas, H4↔PTC member, H6↔pubkey, H7↔signature). Two clauses in `validateGossipPayloadAttestationMessage` are **not** shadowed by the harness:
1. `seenPayloadAttesters.isKnown(epoch, validatorIndex)` gossip-dedup check (lines 53-60) — by design, harness doesn't have gossip semantics.
2. The MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance in `isCurrentSlotGivenGossipDisparity` — harness uses strict equality (H5: `clock.currentSlot !== data.slot`), which is tighter than gossip flow. Negligible practical difference for spec vectors that drive clock via ticks.

Out of scope for #9384, just flagging for follow-up.

### Suggestions for #9384 (judgment call — leaving to you and @markolazic01)

- **Minimal cleanup:** drop H3, H6, H8 (dead/redundant). Keep H1, H4, H5, H7 since they document the spec's `on_payload_attestation_message` semantics that the test runner is asserting. Annotate H5/H7 as "inlined from `validateGossipPayloadAttestationMessage` to match spec's parent function".
- **Status quo:** leave it as-is — twoeths already approved and the harness *does* correctly implement the spec's parent function. The "harness gap" is structural to how `fork_choice/on_payload_attestation_message/` tests are designed (they target `on_payload_attestation_message`, not `notify_ptc_messages` in isolation).
- **Aggressive refactor:** thin the harness to call a single entry point (e.g. spec-equivalent `on_payload_attestation_message` helper) instead of inline. Out of scope for this PR but worth a follow-up issue.

Methodology notes / reproduction: each mutation `git checkout`-restored, `@lodestar/fork-choice` rebuilt between proto-array mutations (per Nico's rule). Tests run with `pnpm vitest run --project spec-minimal test/spec/presets/fork_choice.test.ts -t "on_payload_attestation_message"`. Spec-tests pinned to `v1.7.0-alpha.9` (matches PR HEAD). Mainnet preset not run (minimal-only iteration); all 7 minimal vectors confirmed against the baseline + each mutation. Worktree clean post-pass.

cc @markolazic01 — your A1/A2 comment matched my finding; A3 is just defense-in-depth as you noted in the parallel thread.

**Comment by lodekeeper** [comment]
**Correction to my prior coverage summary** — re-read the spec call chain and the A2 "from-block gap" claim was wrong.

In `specs/gloas/beacon-chain.md` line 1773-1785, `process_payload_attestation` (called during block state-transition via `for_ops(body.payload_attestations, process_payload_attestation)` at line 1523) asserts:
- `data.beacon_block_root == state.latest_block_header.parent_root`
- `data.slot + 1 == state.slot`

So any block body containing a PA with `data.slot != parent.slot` is **rejected at state-transition** before fork-choice's `notify_ptc_messages` (`specs/gloas/fork-choice.md:241`) is even called from `on_block` (line 928).

That means lodestar's A2 (`node.slot !== slot → return` in `protoArray.notifyPtcMessages`) is genuinely defense-in-depth — structurally unreachable on valid inputs through every entry point:
- gossip: `validateGossipPayloadAttestationMessage` pre-validates `block.slot === data.slot`.
- API: same shared `validatePayloadAttestationMessage`.
- from-block: state-transition's `process_payload_attestation` asserts `data.slot + 1 == state.slot` before `importBlock.ts:264` calls `notifyPtcMessages`.

No realistic spec vector can exercise A2 because the spec's own invariants forbid the input. **Dropping the "open a consensus-specs PR for the from-block A2 gap" recommendation** — there's nothing to add upstream.

The other coverage gap (production `validateGossipPayloadAttestationMessage` is not exercised by these spec tests, only shadowed by the harness's inline copy) is still real and tracked as #9455 (cheap unit-test path).

cc @markolazic01 @nflaig — apologies for the noise; should have checked the spec's state-transition assertions before claiming a from-block gap existed.

**Comment by markolazic01** [comment]
Latest two commits are just a conflict resolution.

**Comment by markolazic01** [comment]
Some of the checks I considered redundant affected type safety, got them back and now the workflow should pass.

**Comment by markolazic01** [comment]
Not sure why 2 jobs are failing, but they seem unrelated to the PR changes.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts (hunk 8 above)
Comment (lodekeeper) [reply packages/beacon-node/test/spec/presets/fork_choice.test.ts:275]: Ran the pressure-test yesterday — methodology + per-guard outcomes in `issuecomment-4611833903`, plus a correction on the A2 "from-block gap" claim in `issuecomment-4612620258`.

TL;DR: all 7 `gloas/fork_choice/on_payload_attestation_message/*` vectors are properly exercised by the handler + real-code guards; the only "gap-looking" case turned out to be deliberate spec-level rejection — a PA with `data.slot != parent.slot` is rejected at state-transition by `process_payload_attestation` (`specs/gloas/beacon-chain.md` L1773-1785) before fork-choice's `notify_ptc_messages` ever runs.

**Comment by lodekeeper** [comment]
Re `pullrequestreview-4323816830` — the global-vs-committee index issue gemini flagged was correctly identified and has been fixed: `notifyPtcMessages` now receives committee-relative indices via `getIndicesInPayloadTimelinessCommittee` (returns `number[]`, handles validators occupying multiple committee positions). Verified at HEAD `616cad6`.

**Review by nflaig — APPROVED** [review]
Looks great, thanks @markolazic01 for the thoroughness!

**Review by lodekeeper — APPROVED** [review]
LGTM from my side — final pass on post-cleanup diff:

- Harness shape matches what we landed on after the systematic check-removal pass: load-bearing checks (slot-match gate mirroring impl A2, gossip-flow clock + signature validation) kept; redundant defensive checks dropped.
- Earlier A2 "from-block spec gap" walk-back ([comment](https://github.com/ChainSafe/lodestar/pull/9384#issuecomment-4612620258)) still stands — `process_payload_attestation` state-transition invariant (`data.slot + 1 == state.slot`) means there's nothing to add upstream.
- Follow-up coverage on production `validateGossipPayloadAttestationMessage` tracked as #9455, out of scope here.

CI green, mergeable. Good to ship.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.44.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.44.0) :tada:
